### PR TITLE
Feature pmd

### PR DESCRIPTION
**Pull request by PavelApis**

### CODE REVIEW

File: config/pmd/pmd.xml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    <description>
+        PMD configuration
+    </description>
+
+    <rule ref="category/java/bestpractices.xml">
+        <exclude name="MethodReturnsInternalArray"/>
+    </rule>
+
+    <rule ref="category/java/codestyle.xml">
+        <exclude name="CallSuperInConstructor"/>
```
Comment (mfvanek) [inline config/pmd/pmd.xml:12]: Why this rule has beed excluded?

File: config/pmd/pmd.xml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+    <rule ref="category/java/documentation.xml">
+        <exclude name="CommentRequired"/>
+        <exclude name="CommentSize"/>
+        <exclude name="UncommentedEmptyConstructor"/>
+    </rule>
+
+    <rule ref="category/java/errorprone.xml">
+        <exclude name="BeanMembersShouldSerialize"/>
+        <exclude name="MissingSerialVersionUID"/>
```
Comment (mfvanek) [inline config/pmd/pmd.xml:35]: It looks like a bug in your code

File: src/main/java/io/github/pavelapis/minesweeper/Cell.java [hunk 3]
```diff
@@ -10,25 +10,30 @@
  * value: [0-9] - number of mined neighbors; -1 - mine.
  */
 class Cell extends JButton {
-
+    public static final long serialVersionUID = 1;
```
Comment (mfvanek) [inline src/main/java/io/github/pavelapis/minesweeper/Cell.java:13]: This must be private and unique inside your application.
See also https://www.baeldung.com/java-serial-version-uid

File: src/main/java/io/github/pavelapis/minesweeper/Minesweeper.java [hunk 4]
```diff
@@ -11,9 +11,29 @@
...
 public class Minesweeper extends JPanel {
 
-    int sizeX;
-    int sizeY;
-    Cell[][] field;
+    private final int sizeX;
+    private final int sizeY;
+    private Cell[][] field;
+
+    public int getSizeX() {
```
Comment (mfvanek) [inline src/main/java/io/github/pavelapis/minesweeper/Minesweeper.java:18]: Use lombok

**Review by mfvanek — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/pavelapis/minesweeper/Cell.java (hunk 3 above)
Comment (EnKarin) [reply src/main/java/io/github/pavelapis/minesweeper/Cell.java:13]: Field 'serialVersionUID' is never used

File: config/pmd/pmd.xml (hunk 1 above)
Comment (PavelApis) [reply config/pmd/pmd.xml:12]: This rule has conflict with checkstyles configuration, one of them need to be excluded

Comment (mfvanek) [reply config/pmd/pmd.xml:12]: Alright

File: config/pmd/pmd.xml (hunk 2 above)
Comment (PavelApis) [reply config/pmd/pmd.xml:35]: I’ll read today about serialization and return to this conversation later :)

File: src/main/java/io/github/pavelapis/minesweeper/Minesweeper.java (hunk 4 above)
Comment (PavelApis) [reply src/main/java/io/github/pavelapis/minesweeper/Minesweeper.java]: I seted with lombok necessary getters and setters

File: src/main/java/io/github/pavelapis/minesweeper/Cell.java (hunk 3 above)
Comment (PavelApis) [reply src/main/java/io/github/pavelapis/minesweeper/Cell.java]: Fixed that

**Review by mfvanek — APPROVED** [review]

**Pull request merged**